Endace/sumologic.py

```diff
@@ -0,0 +1,110 @@
+# MIT
+
+# Copyright 2024 Endace Technology Limited
+
+# LEGAL NOTICE: This source code:
+
+# a) is copyright (C) to Endace Technology Limited or any of its related or
+# affiliated companies, a New Zealand company, and its suppliers and licensors
+# ("Endace"), 2024. All rights reserved (except as provided in the license terms
+# referred to in clause (b) below); and
+
+# b) is licensed to you on the terms of the MIT license, detailed below.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import requests
+import json
+
+# Version
+version = '1.0'
+
+# Variables for the Endace Probe URL
+hostname = '<probe hostname>'
+timezone = "etc/utc" # See link for list of time zones: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+title = "Pivot%20from%20SumoLogic"
+datasources = 'tag:rotation-file'
+tools = 'trafficOverTime_by_app,conversations_by_ipaddress'
+url = f"https://{hostname}/vision2/pivotintovision/?title={title}&datasources={datasources}&incidenttime=TIME&reltime=5m&tools={tools}&ip_conv=SOURCEIP%26DESTIP"
+
+# This should be updated to your access id / key. 
+access_id = '<access id>'
+access_key = '<access key>'
+
+# The API endpoint will depend on where your instance is located - in this case we are using US1. You can see more info at https://help.sumologic.com/docs/api/
+api_endpoint_post_extraction_rules = "https://api.sumologic.com/api/v1/extractionRules"
```

Comment:
this url will change depending upon the deployment refer https://help.sumologic.com/docs/api/getting-started/

Reply:
@asimonsendace - I would consider templating this like you did the access key so people don't ignore/miss this detail. Then just have them add it in. @himanshu219 - Would that be the approach you're looking for, or is the current approach sufficient?

Reply:
yes correct